hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/FlinkWriteHelper.java

```diff
@@ -91,13 +91,14 @@ public static FlinkWriteHelper newInstance() {
   @Override
   public List<HoodieRecord<T>> deduplicateRecords(
       List<HoodieRecord<T>> records, HoodieIndex<?, ?> index, int parallelism) {
+    final boolean hasInsert = records.get(0).getCurrentLocation().getInstantTime().equals("I");
```

Comment:
thanks for your contribution. would you add a test for cow table so we can reproduce?